### PR TITLE
Fix gas estimation with configured account

### PR DESCRIPTION
**Pull request by sideninja**
## Description
Use a hardcoded account for gas estimation.
______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -501,32 +501,40 @@ func (b *BlockChainAPI) EstimateGas(
...
 	}
 
-	eoa := newEOATestAccount(b.config.COAKey.String()[2:])
-	// Provide a high enough gas for the tx to be able to execute
-	gas := uint64(15_000_000)
+	// provide a high enough gas for the tx to be able to execute
+	defaultGasLimit := uint64(15_000_000)
 	if args.Gas != nil {
-		gas = uint64(*args.Gas)
+		defaultGasLimit = uint64(*args.Gas)
```
Comment (m-Peter) [inline api/api.go:511]: Nice :ok_hand:

File: api/api.go [hunk 2]
```diff
@@ -501,32 +501,40 @@ func (b *BlockChainAPI) EstimateGas(
...
-	txData := eoa.PrepareSignAndEncodeTx(
-		args.To,
+	txData, err := signGasEstimationTx(args.To,
 		data,
 		(*big.Int)(args.Value),
-		gas,
+		defaultGasLimit,
 		(*big.Int)(args.GasPrice),
 	)
+	if err != nil {
```
Comment (m-Peter) [inline api/api.go:520]: Nice one too :100:

File: api/api.go (hunk 1 above)
Comment (sideninja) [reply api/api.go:509]: yeah if thins don't work better to return some default

File: api/api.go [hunk 3]
```diff
@@ -503,30 +503,36 @@ func (b *BlockChainAPI) EstimateGas(
...
+		b.logger.Error().Err(err).Msg("failed to sign transaction for gas estimate")
+		return hexutil.Uint64(0), err
+	}
 
-	gas, err := b.evm.EstimateGas(ctx, txData)
+	estimatedGas, err := b.evm.EstimateGas(ctx, txData)
 	if err != nil {
 		b.logger.Error().Err(err).Msg("failed to estimate gas")
-		return hexutil.Uint64(0), err
+		return hexutil.Uint64(defaultGasLimit), err // return default gas limit
```
Comment (m-Peter) [inline api/api.go:526]: Maybe return `nil` as an error then? As it might confuse clients, if there is an error, they won't use the value I guess.

Comment (sideninja) [reply api/api.go:526]: ohh, yeah you are right, missed it

File: api/api.go [hunk 4]
```diff
@@ -503,30 +503,37 @@ func (b *BlockChainAPI) EstimateGas(
...
 		args.To,
 		data,
 		(*big.Int)(args.Value),
-		gas,
+		defaultGasLimit,
 		(*big.Int)(args.GasPrice),
 	)
+	if err != nil {
+		b.logger.Error().Err(err).Msg("failed to sign transaction for gas estimate")
+		return hexutil.Uint64(0), nil
```
Comment (m-Peter) [inline api/api.go:521]:
```suggestion
		return hexutil.Uint64(0), err
```

File: api/api.go [hunk 5]
```diff
@@ -503,30 +503,37 @@ func (b *BlockChainAPI) EstimateGas(
...
+	}
 
-	gas, err := b.evm.EstimateGas(ctx, txData)
+	estimatedGas, err := b.evm.EstimateGas(ctx, txData)
 	if err != nil {
 		b.logger.Error().Err(err).Msg("failed to estimate gas")
-		return hexutil.Uint64(0), err
+		return hexutil.Uint64(defaultGasLimit), err // return default gas limit
```
Comment (m-Peter) [inline api/api.go:527]:
```suggestion
		return hexutil.Uint64(defaultGasLimit), nil // return default gas limit
```

**Review by m-Peter — APPROVED** [review]
LGTM :rocket:

**Pull request merged**